xwayland/xwm.c

```diff
@@ -186,7 +188,7 @@ static struct wlr_xwayland_surface *xwayland_surface_create(
 		return NULL;
 	}
 
-	wl_list_insert(&xwm->surfaces, &surface->link);
+	wl_list_insert(&xwm->surfaces_in_map_order, &surface->map_link);
```

Comment:
Note for reviewer: I'm pretty sure this logic is wrong. `surfaces_in_map_order` is used to maintain `_NET_CLIENT_LIST`.

EMWH describes `_NET_CLIENT_LIST` as:

> These arrays contain all X Windows managed by the Window Manager. _NET_CLIENT_LIST has initial mapping order, starting with the oldest window.

`xwm_set_net_client_list` filters this list by `surface->mapped`, but there's no hard guarantee that surfaces will be mapped in the order they're created, right?

Reply:
Yes, this sounds correct. So `surfaces_in_map_order` is actually not in map order.

Although it doesn't seem like i3 orders clients by map time in `_NET_CLIENT_LIST` either. So maybe it would be better to keep the current list with its current name...

Reply:
i3 does order by map time, and has a test enforcing this:

https://github.com/i3/i3/blob/9db03797da3cea5dc6898adc79a68ba4523a409c/testcases/t/223-net-client-list.t

I don't feel strongly about this part of the change. I've backed it out and just left a FIXME as a big flag in case someone in the future stumbles upon this code while debugging some issue.

Reply:
Ah, I missed that, thanks